pkg/volume/rbd/rbd_util.go

```diff
@@ -352,31 +368,23 @@ func (util *RBDUtil) AttachDisk(b rbdMounter) (string, error) {
 		}
 	}
 
-	// If rbd-nbd tools are installed, we will try to map as a nbd device first.
-	nbdToolsFound := false
-	_, err = b.exec.Run("modprobe", "nbd")
-	if err != nil {
-		glog.V(5).Infof("rbd-nbd: nbd modprobe failed with error %v", err)
-	} else if _, err := b.exec.Run("rbd-nbd", "--version"); err != nil {
-		glog.V(5).Infof("rbd-nbd: getting rbd-nbd version failed with error %v", err)
-	} else {
-		glog.V(3).Infof("rbd-nbd tools were found. Attempting to map using rbd-nbd")
-		nbdToolsFound = true
-	}
+	// If rbd-nbd tools are found, we will fallback to it should the default krbd driver fail
+	foundRbdNbd := false
 
-	var devicePath string
-	var mapped bool
-	// The code below attempts to determine which, if any driver, has mapped the device.
-	if nbdToolsFound {
-		// Evalute whether this device was mapped with rbd-nbd.
-		devicePath, mapped = waitForPath(b.Pool, b.Image, 1 /*maxRetries*/, true /*useNbdDriver*/)
-	}
+	devicePath, mapped := waitForPath(b.Pool, b.Image, 1, "rbd")
 	if !mapped {
-		// Evalute whether this device was mapped with rbd.
-		devicePath, mapped = waitForPath(b.Pool, b.Image, 1 /*maxRetries*/, false /*useNbdDriver*/)
+		foundRbdNbd := checkRbdNbdTools(b.exec)
+		if foundRbdNbd {
+			devicePath, mapped = waitForPath(b.Pool, b.Image, 1 , "nbd")
+		}
 	}
 
 	if !mapped {
+		_, err = b.exec.Run("modprobe", "rbd")
```

Comment:
I'd like to move this check up above L374 or into a separate method checking for rbd/krbd.

Reply:
It can move below but not above. It only warns which is the current behavior - a method is not appropriate, I think.